controllers/authors.js

```diff
@@ -0,0 +1,39 @@
+const models = require('../models')
+
+const getAllAuthors = async (request, response) => {
+  try {
+    const allAuthors = await models.authors.findAll()
+
+    return response.send(allAuthors)
+  } catch (error) {
+    return response.status(500).send('Could not retrieve authors, please try again.')
+  }
+}
+
+const getAuthorByIdWithNovelsAndGenres = async (request, response) => {
+  try {
+    const { id } = request.params
+    const matchingAuthor = await models.authors.findAll({
+      include: [{
+        include: [{ model: models.genres }],
+        model: models.novels
+      }],
+      where: {
+        [models.Op.or]: [
+          { id },
+          { nameLast: { [models.Op.like]: `%${id.toLowerCase()}%` } }
+        ]
+      }
+
+    })
+
+    return matchingAuthor
```

Comment:
This doesn't actually work because you used `findAll` instead of `findOne`. `findAll` returns an empty array when there is no match and an empty array gets converted to true. You'd have to do `matchAuthor.length` to check for an empty array. But really here you should be using `findOne` and that would fix this.